src/V3DfgSynthesize.cpp

```diff
@@ -0,0 +1,1699 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Convert DfgLogic into primitive operations
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2025 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+//
+// Synthesize DfgLogic vertices in as a graph, as created by V3DfgAstToDfg
+// into primitive vertices.
+//
```

Comment:
Would be good to have maybe 20-50 lines here summarizing all that this code can optimize.
(Unless there's another place already, but I missed it.)